src/BackgroundTask/CronBasedBackgroundTask.php

```diff
@@ -72,4 +74,23 @@ abstract protected function get_event_name();
 	 * @param mixed[] ...$args Args to pass to the process callback.
 	 */
 	abstract public function process( ...$args );
+
+	/**
+	 * Add new interval for cron schedules.
+	 *
+	 * @param array $cron_schedules List of cron schedules.
+	 *
+	 * @return array List of cron schedules.
+	 */
+	public function add_cron_schedules( $cron_schedules ) {
+
+		$cron_schedules = ( ! empty( $cron_schedules ) && is_array( $cron_schedules ) ) ? $cron_schedules : [];
+
+		$cron_schedules[ self::DEFAULT_INTERVAL_EVERY_TEN_MINUTES ] = [
+			'interval' => ( MINUTE_IN_SECONDS * 10 ),
+			'display'  => esc_html__( 'Every 10 minutes', 'amp' ),
```

Comment:
It's not known where this will be rendered, so escaping here is premature:

```suggestion
			'display'  => __( 'Every 10 minutes', 'amp' ),
```